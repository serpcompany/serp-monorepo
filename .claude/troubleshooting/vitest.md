https://vitest.dev/guide/common-errors.html