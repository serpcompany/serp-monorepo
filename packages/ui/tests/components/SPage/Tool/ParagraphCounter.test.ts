import { mockNuxtImport } from '@nuxt/test-utils/runtime';
import { describe, expect, it } from 'vitest';
import SPageToolParagraphCounter from '../../../../components/SPage/Tool/ParagraphCounter.vue';
import ComponentRender from '../../../componentRender';

mockNuxtImport('useSeoMeta', () => () => {});

describe('sPageToolParagraphCounter Snapshot', () => {
  it('renders the default snapshot correctly', async () => {
    const html = await ComponentRender(
      'SPageToolParagraphCounter',
      {},
      SPageToolParagraphCounter,
    );
    expect(html).toMatchSnapshot();
  });
});
