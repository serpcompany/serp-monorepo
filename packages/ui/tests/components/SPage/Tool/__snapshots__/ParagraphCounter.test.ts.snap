// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SPageToolParagraphCounter Snapshot > renders the default snapshot correctly 1`] = `
"<div class="container">
  <div class="px-4 py-10 sm:px-6 sm:py-16 md:px-8 md:py-20">
    <section class="py-20">
      <h1 class="md:text-center">Paragraph Counter</h1>
      <p data-testid="subtitle" class="sm:text-md text-left text-base md:text-center">A free online tool to count the number of paragraphs in text.</p>
    </section><!-- box -->
    <div>
      <div class="relative inline-flex items-center mb-10 w-full"><textarea rows="15" placeholder="Paste your content here" class="w-full rounded-none border-0 placeholder:text-[var(--ui-text-dimmed)] focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 transition-colors px-2.5 py-1.5 text-sm gap-1.5 text-(--ui-text-highlighted) bg-(--ui-bg) ring ring-inset ring-(--ui-border-accented) focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-(--ui-primary)" value=""></textarea></div>
    </div><!-- button --><button type="button" class="rounded-none font-medium inline-flex items-center focus:outline-hidden disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-sm gap-1.5 text-(--ui-bg) bg-(--ui-primary) hover:bg-(--ui-primary)/75 disabled:bg-(--ui-primary) aria-disabled:bg-(--ui-primary) focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-(--ui-primary)">
      <!--v-if-->Submit
      <!--v-if-->
    </button>
    <!--v-if-->
  </div>
</div>"
`;
